foqus_lib/framework/graph/node.py

```diff
@@ -80,6 +96,97 @@ def setCodeStrings(self):
         self.codeString[61] = "Unknow type string"
 
 
+class pymodel_ml_ai(pymodel):
+    def __init__(self, model):
+        pymodel.__init__(self)
+
+        # attempt to retrieve required information from loaded model, and set defaults otherwise
+        self.model = model
+
+        for i in range(np.shape(self.model.inputs[0])[1]):
+            try:
+                input_label = self.model.layers[1].input_labels[i]
+            except:
```

Comment:
Suggestions to make exceptions more specific:

- Use `except MyExceptionType:` instead of "bare `except`
- Add a comment with more details on when that exception might occur (e.g. "this happens if the input labels in the model do not match those stored in the file" (just an example))